realsense2_camera/src/realsense_node_factory.cpp

```diff
@@ -107,24 +107,31 @@ void RealSenseNodeFactory::getDevice(rs2::device_list list)
                 std::vector<std::string> results;
                 ROS_INFO_STREAM("Device with name " << name << " was found.");
                 std::string port_id = parseUsbPort(pn);
-                if (port_id.empty())
+
+                std::string pid_str(dev.get_info(RS2_CAMERA_INFO_PRODUCT_ID));
+                if(pid_str != "DDS")
                 {
-                    std::stringstream msg;
-                    msg << "Error extracting usb port from device with physical ID: " << pn << std::endl << "Please report on github issue at https://github.com/IntelRealSense/realsense-ros";
-                    if (_usb_port_id.empty())
+                    if (port_id.empty())
                     {
-                        ROS_WARN_STREAM(msg.str());
+                        std::stringstream msg;
+                        msg << "Error extracting usb port from device with physical ID: " << pn << std::endl
+                            << "Please report on github issue at https://github.com/IntelRealSense/realsense-ros";
+                        if (_usb_port_id.empty())
+                        {
+                            ROS_WARN_STREAM(msg.str());
+                        }
+                        else
+                        {
+                            ROS_ERROR_STREAM(msg.str());
+                            ROS_ERROR_STREAM("Please use serial number instead of usb port.");
```

Comment:
The user asked for a USB port?
From the code it looks like the code is using it